src/components/Microsoft.DotNet.UpgradeAssistant/Checks/BypassableReadinessCheck.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Checks
+{
+    public abstract class BypassableReadinessCheck : IUpgradeReadyCheck
+    {
+        private readonly ILogger _logger;
+        private readonly UpgradeOptions _upgradeOptions;
+
+        protected BypassableReadinessCheck(ILogger logger, UpgradeOptions upgradeOptions)
```

Comment:
This class probably isn't necessary. Instead, just update CurrentProjectSelectionStep to check UpgradeOptions and only call bypassable ready checks if the bypass flag isn't set.

Then, just implement all ready checks as IUpgradeReadyCheck directly and have the bypassable ones provide descriptions of problematic technology and links to relevant docs in their IsReadyAsync methods.

You could have a base class like this if you want to ensure logging is uniform, but I would only have it do that (standardize logging) and now have it checking whether checks are bypassed or not.

Reply:
Class removed